### PR TITLE
Remove filter propeties from marathon store

### PR DESCRIPTION
**Pull request by aldipower**
Bring the filter properties into a better location.
Separation of concerns.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @daiyiwhale

File: src/js/structs/Tree.js [hunk 1]
```diff
@@ -127,11 +127,17 @@ module.exports = class Tree extends List {
...
 
-      if (filter.name) {
-        let tree =
-          new this.constructor(Object.assign({},
-            this, {items: services, filterProperties: this.getFilterProperties()}));
-        services = tree.filterItemsByText(filter.name).getItems();
+      if (filter.id) {
+        let filterProperties = Object.assign({}, this.getFilterProperties(), {
+          name: function (item) {
+            return item.id;
```
Comment (orlandohohmeier) [inline src/js/structs/Tree.js]: Why are you using the item `id` for the `name` filter?

File: src/js/structs/Tree.js [hunk 2]
```diff
@@ -127,11 +127,17 @@ module.exports = class Tree extends List {
...
+          }
+        });
+
+        let tree = new this.constructor(Object.assign({},
```
Comment (orlandohohmeier) [inline src/js/structs/Tree.js]: You don't need to create a new instance of tree to define the filter properties – just pass them to the `filterItemsByText` as a second argument.

**Comment by orlandohohmeier** [comment]
Not part of the PR, but let's please remove the `filterByFilter` method (legacy code) and use the simple `filterItems` method instead.  If `filterItems` is not sufficient, to do the service filtering, let's please introduces the respective method to the `ServiceTree` instead of the general `Tree` struct.

**Comment by pierlo-upitup** [comment]
@aldipower needs a good ol' rebase

File: src/js/structs/Tree.js (hunk 2 above)
Comment (aldipower) [inline src/js/structs/Tree.js]: Danke.

**Comment by aldipower** [comment]
@pierlo-upitup Gerebased.

@orlandohohmeier Yes, do you know if there is a ticket existent?

File: src/js/structs/Tree.js (hunk 1 above)
Comment (aldipower) [inline src/js/structs/Tree.js]: Because a lot of marathon services doesn't have a name property.

Comment (orlandohohmeier) [inline src/js/structs/Tree.js]: All service instance (frameworks as well as applications) do provide an id (`getId`) and name (`getName`) getter – let's use them.

**Comment by orlandohohmeier** [comment]
@aldipower Probably not... Would you mind creating one?

File: src/js/structs/Tree.js (hunk 1 above)
Comment (aldipower) [inline src/js/structs/Tree.js]: OK, I think I've got what you've meant now.

**Comment by pierlo-upitup** [comment]
Tested against latest unified-ui and works. LGTM.

**Comment by philipnrmn** [comment]
Beat me to the merge :)

**Pull request merged**